File: 0764-largest-plus-sign/0764-largest-plus-sign.cs
public class Solution {
    public int OrderOfLargestPlusSign(int n, int[][] mines) {
        var maxOrder = 0;

        var isMine = new bool[n,n];
        foreach(var mine in mines)
            isMine[mine[0],mine[1]] = true;

        // store max possible 1s in any direction
        var dp = new int[n,n];
        for(var i=0; i<n; i++){
            // traverse from left to right
            var count=0;
            for(var j=0; j<n; j++){
                count = isMine[i,j] ? 0 : count+1;
                dp[i,j] = count;
            }

            // traverse from right to left
            count=0;
            for(var j=n-1; j>=0; j--){
                count = isMine[i,j] ? 0 : count+1;
                dp[i,j] = Math.Min(dp[i,j], count);
            }
        }

        for(var i=0; i<n; i++){
            // traverse from up to down
            var count=0;
            for(var j=0; j<n; j++){
                count = isMine[j,i] ? 0 : count+1;
                dp[j,i] = Math.Min(dp[j,i], count);
            }

            // traverse from down to up
            count=0;
            for(var j=n-1; j>=0; j--){
                count = isMine[j,i] ? 0 : count+1;
                dp[j,i] = Math.Min(dp[j,i], count);
                
                maxOrder = Math.Max(maxOrder, dp[j,i]);
            }
        }

        return maxOrder;
    }
}

File: 0764-largest-plus-sign/0764-largest-plus-sign.py
class Solution:
    def orderOfLargestPlusSign(self, n: int, mines: List[List[int]]) -> int:
        maxOrder = 0

        isMine = [[False]*n for _ in range(n)]
        for mine in mines:
            isMine[mine[0]][mine[1]] = True

        # store max possible 1s in any direction
        dp = [[0]*n for _ in range(n)]
        for i in range(n):
            # traverse from left to right
            count=0
            for j in range(n):
                count = 0 if isMine[i][j] else count+1
                dp[i][j] = count

            # traverse from right to left
            count=0
            for j in range(n-1, -1, -1):
                count = 0 if isMine[i][j] else count+1
                dp[i][j] = min(dp[i][j], count)

        for i in range(n):
            # traverse from up to down
            count=0
            for j in range(n):
                count = 0 if isMine[j][i] else count+1
                dp[j][i] = min(dp[j][i], count)

            # traverse from down to up
            count=0
            for j in range(n-1, -1, -1):
                count = 0 if isMine[j][i] else count+1
                dp[j][i] = min(dp[j][i], count)
                
                maxOrder = max(maxOrder, dp[j][i])

        return maxOrder